### PR TITLE
Renaming CUDA Quantum to CUDA-Q

### PR DESCRIPTION
**Pull request by bettinaheim**
Also fixes https://github.com/NVIDIA/cuda-quantum/issues/1351

### CODE REVIEW

**Review by efratshabtai — APPROVED** [review]

File: docker/release/cudaq.Dockerfile [hunk 1]
```diff
@@ -89,7 +89,7 @@ RUN echo "$CUDA_QUANTUM_PATH" > /usr/local/lib/python$(python --version | egrep
 # Some tools related to shell handling.
 
 ARG COPYRIGHT_NOTICE="=========================\n\
-   NVIDIA CUDA Quantum   \n\
+   NVIDIA CUDA-Q   \n\
```
Comment (bmhowe23) [inline docker/release/cudaq.Dockerfile:92]: Nit: re-centering:
```suggestion
      NVIDIA CUDA-Q      \n\
```

File: docs/sphinx/conf.py [hunk 2]
```diff
@@ -23,7 +23,7 @@
 
 # -- Project information -----------------------------------------------------
 
-project = 'NVIDIA CUDA Quantum'
+project = 'NVIDIA CUDA-Q'
 copyright = '2023, NVIDIA Corporation & Affiliates'
```
Comment (bmhowe23) [inline docs/sphinx/conf.py:27]: Just happened to notice that this since your change was right above it ... this still shows 2023 copyright.

**Review by bmhowe23 — APPROVED** [review]
LGTM ... just one nit about formatting for banner shown from the Docker image, and one likely-out-of-bounds nit about a lingering 2023 copyright date.

**Review by schweitzpgi — APPROVED** [review]
Skimmed and spot checked.

**Comment by bmhowe23** [comment]
> Just FYI, There are still some instances of "CUDA Quantum" in the repository https://github.com/search?q=repo%3Asacpis%2Fcuda-quantum+%22CUDA+Quantum%22&type=code
> 
> May be it is safe to ignore. Just thought of highlighting it.

It might be better to look at the hits in this link (NVIDIA repo): https://github.com/search?q=repo%3ANVIDIA%2Fcuda-quantum+%22CUDA+Quantum%22&type=code

I'm not sure why it is different, but it is.

**Pull request merged**